src/bsk_rl/sats/satellite.py

```diff
@@ -0,0 +1,300 @@
+                will stop the simulation.
+        """
+        self.name = name
+        self.logger = logging.getLogger(__name__).getChild(self.name)
+        if sat_args is None:
+            sat_args = self.default_sat_args()
+        self.sat_args_generator = self.default_sat_args(**sat_args)
+        self.simulator: "Simulator"
+        self.fsw: "fsw.FSWModel"
+        self.dynamics: "dyn.DynamicsModel"
+        self.data_store: "DataStore"
+        self.requires_retasking: bool
+        self.variable_interval = variable_interval
+        self._timed_terminal_event_name = None
+        self.observation_builder = ObservationBuilder(self, obs_type=obs_type)
+        self.action_builder = select_action_builder(self)
+
+    @property
+    def id(self) -> str:
+        """Unique human-readable identifier."""
+        return f"{self.name}_{id(self)}"
+
+    def _generate_sat_args(self) -> None:
+        """Instantiate sat_args from any randomizers in provided sat_args."""
+        self.sat_args = {
+            k: v if not callable(v) else v() for k, v in self.sat_args_generator.items()
+        }
+        self.logger.debug(f"Satellite initialized with {self.sat_args}")
+
+    def reset_pre_sim(self) -> None:
+        """Called during environment reset, before Basilisk simulation initialization."""
+        self.info = []
+        self.requires_retasking = True
+        self._generate_sat_args()
+        self.trajectory = TrajectorySimulator(
+            utc_init=self.sat_args["utc_init"],
+            rN=self.sat_args["rN"],
+            vN=self.sat_args["vN"],
+            oe=self.sat_args["oe"],
+            mu=self.sat_args["mu"],
+        )
+        self._timed_terminal_event_name = None
+
+    def set_simulator(self, simulator: "Simulator"):
+        """Set the simulator for models.
+
+        Called during simulator initialization.
+
+        Args:
+            simulator: Basilisk simulator
+
+        :meta private:
+        """
+        self.simulator = proxy(simulator)
+
+    def set_dynamics(self, dyn_rate: float) -> "dyn.DynamicsModel":
+        """Create dynamics model; called during simulator initialization.
+
+        Args:
+            dyn_rate: rate for dynamics simulation [s]
+
+        Returns:
+            Satellite's dynamics model
+
+        :meta private:
+        """
+        dynamics = self.dyn_type(self, dyn_rate, **self.sat_args)
+        self.dynamics = proxy(dynamics)
+        return dynamics
+
+    def set_fsw(self, fsw_rate: float) -> "fsw.FSWModel":
+        """Create flight software model; called during simulator initialization.
+
+        Args:
+            fsw_rate: rate for FSW simulation [s]
+
+        Returns:
+            Satellite's FSW model
+
+        :meta private:
+        """
+        fsw = self.fsw_type(self, fsw_rate, **self.sat_args)
+        self.fsw = proxy(fsw)
+        return fsw
+
+    def reset_post_sim(self) -> None:
```

Comment:
I wonder if changing this to something such as `reste_post_sim_init` could make it more explicit. But now we have the documentation explaining when it is called, so not so necessary

Reply:
Still debating whether this is a change worth making. I think docs describe it pretty well and it's rarely something a front end user would touch.

Reply:
Decided to go with your suggestion. Will push that